File: v-lang/lang.java/MQ/v-kafka/src/main/java/org/example/Producer.java
package org.example;

import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerRecord;

import java.util.Properties;

public class Producer {
    public static void main(String[] args) {
        Properties props = new Properties();
        props.put("bootstrap.servers", "localhost:9092");
        props.put("acks", "all");
        props.put("retries", 0);
        props.put("batch.size", 16384);
        props.put("linger.ms", 1);
        props.put("buffer.memory", 33554432);
        props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
        props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");

        org.apache.kafka.clients.producer.Producer<String, String> producer = new KafkaProducer<>(props);
        for (int i = 0; i < 100; i++) {
            producer.send(new ProducerRecord<>("flink_test", Integer.toString(i), Integer.toString(i)));
        }

        producer.close();
    }
}

File: v-lang/l-kotlin/v-message/v-kafka/src/main/kotlin/Producer.kt
package org.example

import org.apache.kafka.clients.producer.KafkaProducer
import org.apache.kafka.clients.producer.ProducerConfig
import org.apache.kafka.clients.producer.ProducerRecord
import org.apache.kafka.common.serialization.StringSerializer
import java.util.*

fun main() {
    val props = Properties().apply {
        put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9094")
        put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer::class.java)
        put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer::class.java)
    }

    val producer = KafkaProducer<String, String>(props)
    val record = ProducerRecord("vitamin", "key", "Hello, hutao")
    producer.send(record) { metadata, exception ->
        if (exception != null) {
            println("Error while producing message: ${exception.message}")
        } else {
            println("Sent message to partition ${metadata?.partition()} with offset ${metadata?.offset()}")
        }
    }
    producer.close()
}